igor-web/src/main/java/com/netflix/spinnaker/igor/artifacts/ArtifactService.java

```diff
@@ -33,9 +33,26 @@
 
   /** Used to populate the manual trigger dropdown with options */
   @Nonnull
-  List<String> getArtifactVersions(@Nonnull String name, String releaseStatus);
+  List<String> getArtifactVersions(
+      @Nonnull String type, @Nonnull String name, String releaseStatus);
 
   /** Used to fetch a specific artifact for decorating a trigger */
   @Nonnull
-  Artifact getArtifact(@Nonnull String name, @Nonnull String version);
+  Artifact getArtifact(@Nonnull String type, @Nonnull String name, @Nonnull String version);
```

Comment:
In the future I'd prefer if you added a new argument as the last parameter to this function. No need to change here though.

Reply:
Can you elaborate on why? I added it to that position specifically because it made more logical sense to me that you'd specify the type before the name.

Reply:
Groovy doesn't always tell you when you break function calls, because groovy sucks. We have a lot of groovy, so I'm just always cautious (not applicable here, but applicable elsewhere). Also, I think it's easy to mix up test parameters when you add at the beginning.

Reply:
ACK.